src/op/op_util.cc

```diff
@@ -206,6 +206,34 @@ Expr ReplaceTensor(Expr expr,
 }
 
 
+void ReplaceTensorRecursivelyImpl(Tensor tensor,
+                                  std::unordered_map<Tensor, Tensor>* replace) {
```

Comment:
Should we use `std::unordered_map<Tensor, Tensor, NodeHash, NodeEqual>*`?

Reply:
I don't think so. The Tensor class has its own coarser notion of equality and it has its own specialization of `std::hash`. I think using the ordinary reference equality may cause problems even for this particular function.